.nycrc

```diff
@@ -0,0 +1,22 @@
+{
+  "all": true,
+  "branches": 95,
+  "lines": 95,
+  "check-coverage": true,
+  "compact": false,
+  "exclude": [
+    "public/assets/frontend/component_guide",
+    "public/assets/frontend/govuk_publishing_components*",
+    "public/assets/frontend/manifest-*",
+    "public/assets/frontend/application-*",
+    "public/assets/frontend/test-dependencies-*",
+    "public/assets/frontend/dependencies-*",
+    "public/assets/frontend/es6-components-*"
+  ],
+  "report-dir": "coverage/javascript",
+  "reporter": [
+    "text",
+    "html"
```

Comment:
I wonder if we should remove the `html` option from the base level config for now for the reasons below:

- It appears that the HTML report will also be created when running the Continuous Integration checks but is not required here
- We can always add this feature in the future for use in a development environment by creating a custom nyc dev config - https://github.com/istanbuljs/nyc?tab=readme-ov-file#adding-your-overrides
- Would allow us to remove some of the code used in the nyc-test-coverage-reporter file used to delete the reports - https://github.com/alphagov/frontend/pull/4142/files#diff-cad9e4a78533bc33cf9c9b72531808b7be8503ad13528c114c664067d243893cR27

Reply:
No, continuous integration runs jasmine:ci

Also, developers need html reports in order to check the coverage, otherwise it's difficult to discover what tests needs to be added.

Reply:
> No, continuous integration runs jasmine:ci

`yarn run jasmine:prepare && yarn run jasmine-browser-runner runSpecs && yarn run nyc:report`

This script also appears to run `nyc:report`, at which point does the HTML report also get generated?

Reply:
Yes, but nyc:report only runs the reporters specified in this array, so no html report will be generated. You can check yourself by commenting `"html"`, removing the existing report and running `yarn run jasmine:ci` again.

Reply:
After the discussion we decided to add a separate package.json script for creating the html report and require the user to manually remove the previous report before generating a new one.

Reply:
Done

Reply:
Thanks, I think we can always look to add another step / something in the build process to help improve the developer experience in the future to avoid manually deleting the files generated by the HTML report